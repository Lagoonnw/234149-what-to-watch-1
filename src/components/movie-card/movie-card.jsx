import React from 'react';
import PropTypes from 'prop-types';

export const MovieCard = (props) => {
  const {name, imgSrc, onClick} = props;
  const clickHandler = () => onClick({name, imgSrc});

  return (
    <article className="small-movie-card catalog__movies-card">
      <button className="small-movie-card__play-btn" type="button" onClick={clickHandler}>Play</button>
      <div className="small-movie-card__image">
        <img src={imgSrc} alt={name} width="280" height="175"/>
      </div>
      <h3 className="small-movie-card__title">
        <a className="small-movie-card__link" href="movie-page.html">{name}</a>
      </h3>
    </article>
  );
};

MovieCard.propTypes = {
  name: PropTypes.string.isRequired,
  imgSrc: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired
};
