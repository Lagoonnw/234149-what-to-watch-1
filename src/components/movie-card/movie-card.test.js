import React from 'react';
import renderer from 'react-test-renderer';
import {MovieCard} from './movie-card.jsx';

describe(`MovieCard render`, () => {
  test(`Should render movie card`, () => {
    const mock = {
      name: `Need Mass Effect movie`,
      imgSrc: ``,
      onMouseEnter: jest.fn(),
      onClick: jest.fn()
    };
    const tree = renderer.create(<MovieCard {...mock}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
