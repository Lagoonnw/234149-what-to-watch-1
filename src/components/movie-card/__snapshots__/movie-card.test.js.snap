// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard render Should render movie card 1`] = `
<article
  className="small-movie-card catalog__movies-card"
>
  <button
    className="small-movie-card__play-btn"
    onClick={[Function]}
    type="button"
  >
    Play
  </button>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="Need Mass Effect movie"
      height="175"
      src=""
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      Need Mass Effect movie
    </a>
  </h3>
</article>
`;
